flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -470,7 +470,8 @@ public static String[] mergeListsToArray(List<String> base, List<String> append)
             ConfigOptions.key("parallelism.default")
                     .intType()
                     .defaultValue(1)
-                    .withDescription("Default parallelism for jobs.");
+                    .withDescription(
+                            "Default parallelism for jobs. There are two special cases. In the first case, when creating a StreamExecutioneEnvironment without specifying parallelism.default, the number of processors available to the Java virtual machine will be used. In the second case, when creating mini cluster without specifying parallelism.default, the total number of slots in mini cluster will be used.");
```

Comment:
I have a question about the first case, does it cover all running modes? 

For application and session mode, and standalone, yarn or kubernetes type. Doesn't `parallelism.default` take effect for all of these cases?

Reply:
No. The first case only found in the local mode.